Marlin/src/inc/Conditionals_post.h

```diff
@@ -543,10 +543,11 @@
   #endif
 
   #if DISABLED(USB_FLASH_DRIVE_SUPPORT) || BOTH(MULTI_VOLUME, VOLUME_SD_ONBOARD)
+    #if ENABLED(SDSUPPORT)
+      #define NEED_SD2CARD_SPI 1
+    #endif
```

Comment:
The `SDSUPPORT` option is used as the flag for any and all media types, not just SD Cards. If `SDSUPPORT` is not enabled, none of the file access code (e.g., `cardreader.cpp`) will be compiled in. So the code needs to be reworked a lot more if the intent is to make `SDSUPPORT` only refer to SPI-based SD cards.

Since it is possible to have both onboard and external media drives, we would also need to rework the code to remove the assumption that there is only one SD card, either `ONBOARD`, `LCD`, or `CUSTOM_CABLE`.

Reply:
We could have SD_STORAGE, SDIO_STORAGE, and USBFLASH_STORAGE options underneath the requires SDSUPPORT. I see that there are currently 3 media_driver_X types implemented:

1. media_driver_usbFlash
2. media_driver_sdcard
3. media_driver_sdiocard

Even though the location of the SD slots may be onboard or on LCD to the user, the access method for the SD slots can vary between all three methods no matter where the slots physically are.

Reply:
The current assumptions are probably a little constrained. As I've been working on separating out the LCD pins in another PR I've become better acquainted with the way that SD Card pins are allocated, either with or without an LCD controller. Here we have a number of different possibilities.

- Onboard SD Card drive with dedicated pins, sometimes but not always the native Hardware SPI.
- Onboard SD Card drive with SDIO, so the card can be released by Marlin and seen by the host PC over USB.
  - I'm not sure whether SPI-based SD Cards can be shared in this way.
- No onboard SD Card drive, but the LCD has one, usually sharing SPI pins with the (SPI-based) LCD.
- No onboard SD Card drive, but an SD Card can be added. With native Hardware SPI or Software SPI.
- Onboard USB Flash Drive is rare, but I have a board or two with one. Presumably Hardware SPI.
  - I'm not sure whether any board has SDIO Flash Drive, or if SDIO is a thing for Flash Drives.
  - I'm not sure whether SPI or SDIO Flash Drives can be released and seen by the host PC over USB.

Right now pins files tend to follow the configuration of the original machine they came with, so if there is an LCD Controller with an SD Card drive, the pins file assumes that the presence of an LCD indicates the use of the SD Card drive on the LCD. Most pins files allow that to be overridden.

The "Custom Cable" option for SD Card drives is handled by very few pins files, and it could be better. If you define your own SPI pins (with `SD_` prefix) and use `CUSTOM_CABLE` then pins files should allow it, but the logic is not there.

There is a lot of overlap between the allocation of pins for the LCD and SD, so it might be good to tackle these two problems at the same time or in a similar manner. See #25650 for that little project, which is coming along slowly but steadily. There I'm trying to weed out assumptions about whether Software or Hardware SPI is used, but also trying to see if that can be based on the chosen pins. i.e., If the pins are not the standard SPI0, SPI1, etc., then we can assume Software-based SPI for that drive.